### PR TITLE
upload artifact per build for easier review

### PR DESCRIPTION
**Pull request by jbkoh**

### CODE REVIEW

File: .github/workflows/builds.yml [hunk 1]
```diff
@@ -39,6 +36,14 @@ jobs:
...
         python generate_shacl.py
         cd ..
+    - name: upload ttls
+      if: ${{ matrix.python-version }} == "3.7.9"
+      uses: actions/upload-artifact@v3
+      with:
+        name: ontology
+        path: |
+          Brick.ttl
+          shacl/BrickShape.ttl
```
Comment (gtfierro) [inline .github/workflows/builds.yml:46]: we don't need to include the `BrickShape.ttl` file ; it should be included in the `Brick.ttl` build.

**Review by gtfierro — COMMENTED** [review]
Thanks for the PR! This is a nice idea. Do you know where the artifacts get uploaded?

File: .github/workflows/builds.yml [hunk 2]
```diff
@@ -6,9 +6,6 @@ on:
         - master
         - v1.*
   pull_request:
-    branches:
-        - master
-        - v1.*
```
Comment (gtfierro) [inline .github/workflows/builds.yml:11]: Does leaving this empty mean that it triggers on *all* PRs?

Comment (jbkoh) [reply .github/workflows/builds.yml:11]: Yeah, that is intended. Every PR deserves a build and automated validation. Are there reasons not to do so?

**Comment by jbkoh** [comment]
> Thanks for the PR! This is a nice idea. Do you know where the artifacts get uploaded?

You can find artifacts at the Summary of each Action. E.g.,
image

File: .github/workflows/builds.yml (hunk 2 above)
Comment (gtfierro) [reply .github/workflows/builds.yml:11]: I agree with you that we want a build on all PRs... my question is whether or not this was the correct GH action syntax to accomplish that

File: .github/workflows/builds.yml [hunk 3]
```diff
@@ -39,6 +36,13 @@ jobs:
         cd shacl
         python generate_shacl.py
         cd ..
+    - name: upload ttls
+      if: ${{ matrix.python-version }} == "3.7.9"
+      uses: actions/upload-artifact@v3
+      with:
+        name: ontology
```
Comment (gtfierro) [inline .github/workflows/builds.yml:43]: Can we change the name of the artifact to "Brick-files" or something more descriptive and Brick-specific than "ontology"? "Brick-ontology" would be fine as well

File: .github/workflows/builds.yml (hunk 2 above)
Comment (jbkoh) [reply .github/workflows/builds.yml:11]: Sorry, misunderstood. Yes, it triggers on all PRs whose activit type is one of opened, synchronize, or reopened.

> Note: More than one activity type triggers this event. For information about each activity type, see "[Webhook events and payloads](https://docs.github.com/en/github-ae@latest/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#pull_request)." By default, a workflow only runs when a pull_request event's activity type is opened, synchronize, or reopened. You can specify different activity types using the types keyword. For more information, see "[Workflow syntax for GitHub Actions](https://docs.github.com/en/github-ae@latest/articles/workflow-syntax-for-github-actions#onevent_nametypes)."

**Comment by jbkoh** [comment]
@gtfierro I just updated the name. You were right that artifacts are always zipped at the UI: https://github.com/actions/upload-artifact#zipped-artifact-downloads, though those files are not zipped in their system. I updated the filename. Please let me know if it's ready to be merged.

**Comment by gtfierro** [comment]
Looks great! I"ll merge it in

**Comment by gtfierro** [comment]
I'm not questioning whether this is a good idea; I agree with you this is what we want to do. My question is whether or not this is the correct syntax for accomplishing having a build on every pull request. I would think there would be some sort of glob pattern, rather than just leaving the trigger blank?


________________________________
From: Jason B. Koh ***@***.***>
Sent: Thursday, April 7, 2022 11:05:38 PM
To: BrickSchema/Brick ***@***.***>
Cc: Gabriel Fierro ***@***.***>; Review requested ***@***.***>
Subject: [External] Re: [BrickSchema/Brick] upload artifact per build for easier review (PR #388)


@jbkoh commented on this pull request.

________________________________

In .github/workflows/builds.yml<https://nam04.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FBrickSchema%2FBrick%2Fpull%2F388%23discussion_r845739999&data=04%7C01%7Cgtfierro%40mines.edu%7Cbda54bba843144722ca408da191d6c35%7C997209e009b346239a4d76afa44a675c%7C0%7C0%7C637849911419915170%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000&sdata=IJ27nLfUsrr0Fi1USmWVDKVgocbMLDXYJE885wipU9I%3D&reserved=0>:

> -    branches:
-        - master
-        - v1.*


Yeah, that is intended. Every PR deserves a build and automated validation. Are there reasons not to do so?

**Comment by jbkoh** [comment]
I believe the syntax is correct, but I can take a look later; I'm traveling now. Have you seen unexpected behaviors?

**Pull request merged**